packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -80,223 +87,269 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
+    const itemResponse = response?.item?.filter((i) => i.linkId === item.linkId) ?? [];
+
+    const repeatedItem =
+      item.type === QuestionnaireItemType.group ? (
+        <QuestionnaireRepeatedGroup
+          key={item.linkId}
+          item={item}
+          response={itemResponse}
+          onChange={onChange}
+          checkForQuestionEnabled={checkForQuestionEnabled}
+        />
+      ) : (
+        <QuestionnaireRepeatableItem
+          key={item.linkId}
+          item={item}
+          response={itemResponse[0]}
+          onChange={onChange}
+          checkForQuestionEnabled={checkForQuestionEnabled}
+        />
+      );
 
-  const questionForm = props.items.map((item, index) => {
-    if (props.renderPages) {
+    if (renderPages) {
       return (
-        <Stepper.Step label={item.text} key={item.linkId}>
-          <QuestionnaireFormArrayContent
-            key={`${item.linkId}-${index}`}
-            item={item}
-            index={index}
-            allResponses={props.allResponses}
-            currentResponseItems={currentResponseItems}
-            groupSequence={props.groupSequence}
-            setResponseItem={setResponseItem}
-          />
+        <Stepper.Step key={item.linkId} label={item.text}>
+          {repeatedItem}
         </Stepper.Step>
       );
     }
-    return (
-      <QuestionnaireFormArrayContent
-        key={`${item.linkId}-${index}`}
-        item={item}
-        index={index}
-        groupSequence={props.groupSequence}
-        allResponses={props.allResponses}
-        currentResponseItems={currentResponseItems}
-        setResponseItem={setResponseItem}
-      />
-    );
+    return repeatedItem;
   });
 
-  if (props.renderPages) {
-    return (
-      <Stepper active={props.activePage ?? 0} allowNextStepsSelect={false} p={6}>
-        {questionForm}
-      </Stepper>
-    );
-  }
-  return <Stack>{questionForm}</Stack>;
+  return (
+    <>
+      {renderPages && (
+        <Stepper active={activePage ?? 0} allowNextStepsSelect={false} p={6}>
+          {form}
+        </Stepper>
+      )}
+      {!renderPages && <Stack>{form}</Stack>}
+      <ButtonGroup
+        activePage={activePage ?? 0}
+        numberOfPages={numberOfPages}
+        nextStep={nextStep}
+        prevStep={prevStep}
+        submitButtonText={submitButtonText}
+      />
+    </>
+  );
 }
 
-interface QuestionnaireFormArrayContentProps {
-  item: QuestionnaireItem;
-  index: number;
-  allResponses: QuestionnaireResponseItem[];
-  currentResponseItems: QuestionnaireResponseItem[];
-  groupSequence?: number;
-  setResponseItem: (responseId: string, newResponseItem: QuestionnaireResponseItem) => void;
+interface QuestionnaireRepeatableGroupProps {
+  readonly item: QuestionnaireItem;
+  readonly response: QuestionnaireResponseItem[];
+  readonly checkForQuestionEnabled: (item: QuestionnaireItem) => boolean;
+  readonly onChange: (responses: QuestionnaireResponseItem | QuestionnaireResponseItem[]) => void;
```

Comment:
Is there a reason we need both single and array here?

Reply:
This is for a group nested inside another group 

e.g 
```
{
   type: group,
   text: groupOuter,
   linkId: groupOuter
   repeats: true, 
   item: [
         {
               type: group,
               text: groupInner,
               linkedId: groupInner,
               repeats: true, 
               item: [
                   { type: string,
                    text: string,
                    linkId: string
                  }
              ]
          }
     ]
}
              

```